lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -790,7 +794,11 @@ tokenize_number([H | T], Acc, Length, Bool) when ?is_digit(H) ->
 
 %% Cast to float...
 tokenize_number(Rest, Acc, Length, true) ->
-  {Rest, list_to_float(lists:reverse(Acc)), Length};
+  try
+    {Rest, list_to_float(lists:reverse(Acc)), Length}
+  catch
+    error:badarg -> {error, "float number out of range ", lists:reverse(Acc)}
```

Comment:
Looking at the tokenizer its possible to have invalid floats on argument error, not just out of range, e.g "1.1e1e1".  Should expand on the message to `float number out of range or invalid`?